nncf/experimental/torch/nas/bootstrapNAS/elasticity/elastic_width.py

```diff
@@ -0,0 +1,891 @@
+        """
+        Constructor.
+
+        :param max_width: maximum number of output channels in the original operation.
+        :param node_name: string representation of operation address. It's used for more informative messages only.
+        :param elastic_width_params: parameters to configure elastic width for the operation.
+        """
+        super().__init__(max_width=max_width, node_name=node_name)
+        if elastic_width_params is None:
+            elastic_width_params = {}
+        min_width = elastic_width_params.get('min_width', 32)
+        max_num_params = elastic_width_params.get('max_num_widths', -1)
+        width_step = elastic_width_params.get('width_step', 32)
+        width_multipliers = elastic_width_params.get('width_multipliers', [])
+
+        self._width_list = self._generate_width_list(self._max_width, min_width, max_num_params, width_step,
+                                                     width_multipliers)
+
+    @property
+    def width_list(self) -> List[int]:
+        """
+        list of all available widths to select from. Each value corresponds to a single element in the search space of
+        operation. The search space of the model is cartesian product of search spaces of operation.
+        If all widths starting from 1 to maximum number of channels with step size 1 are available, the search space
+        would be prohibitively large to efficiently train and search.
+
+        That's why there are elastic width parameters that constraint number of all available widths.
+
+        :return: list of widths
+        """
+        return self._width_list
+
+    def set_active_width(self, width: int) -> None:
+        """
+        Sets current level of elasticity for the operation - number of output channels to trim.
+        The actual trimming of specified number of channels happens on forward call.
+        The value should be less the original width and more than one. Zero number of channels is
+        supported through Dynamic Depth feature.
+
+        :param width: number of output channels
+        """
+        if width not in self.width_list and width != self.max_width:
+            raise ValueError(f'Invalid number of output channels to set: {width} in scope={self._node_name}. '
+                             f'Should be a number in {self.width_list}')
+        super().set_active_width(width)
+
+    @staticmethod
+    def _generate_width_list(max_width: int, min_width: int, max_num_params: int, width_step: int,
+                             width_multipliers: List[float]) -> List[int]:
+        """
+        Generates list of available width values.
+
+        :param max_width: maximum value of width
+        :param min_width: minimum value of width
+        :param max_num_params: maximum number of generated width values.
+        :param width_step: step size to select widths starting from minimal width to maximum one.
```

Comment:
As far as I understand, in case of `width_multipliers` is `None` list is constructed by iteratively substracting `width_step` from `max_width` param e.g. `[max_width - i]` where i changes in range to satisfy the constraints. For me description below is more appropriate
```suggestion
        :param width_step: step size to select widths starting from maximum width to minimum one.
```

Reply:
Thanks! I've paraphrased it to make it more clear.